src/native/corehost/apphost/apphost.windows.cpp

```diff
@@ -86,7 +89,52 @@ namespace
         return msg;
     }
 
-    void show_error_dialog(const pal::char_t *executable_name, int error_code)
+    class activation_context final
+    {
+    public:
+        activation_context(const pal::char_t* source, const pal::char_t* resource_name)
+            : m_activated{false}
+            , m_context_handle{INVALID_HANDLE_VALUE}
+            , m_cookie{}
+        {
+            ACTCTXW actctx = { sizeof(ACTCTXW), 0, source };
+            if (resource_name != nullptr)
+            {
+                actctx.lpResourceName = resource_name;
+                actctx.dwFlags = ACTCTX_FLAG_RESOURCE_NAME_VALID;
+            }
+
+            m_context_handle = ::CreateActCtxW(&actctx);
+            if (m_context_handle == INVALID_HANDLE_VALUE)
+                return;
+
+            if (::ActivateActCtx(m_context_handle, &m_cookie) == FALSE)
+                return;
+
+            m_activated = true;
+        }
+
+        ~activation_context()
```

Comment:
technically we don't need this - the process will exit anyway. (could help with size)